frb_codegen/src/ir/ty_delegate.rs

```diff
@@ -66,6 +86,55 @@ impl IrTypeDelegate {
             }),
         }
     }
+
+    pub fn inner_dart_api_type(&self) -> String {
+        match self {
+            IrTypeDelegate::GeneralArray { general, .. } => general.dart_api_type(),
+            IrTypeDelegate::PrimitiveArray { primitive, .. } => {
+                use IrTypePrimitive::*;
+                match &primitive {
+                    U8 | I8 | U16 | I16 | U32 | I32 | Usize => "int",
+                    U64 | I64 => "BigInt",
+                    F32 | F64 => "double",
+                    Bool => "bool",
+                    Unit => "void",
+                }
+                .to_string()
+            }
+            _ => panic!("Unexpected type"),
+        }
+    }
+    pub fn inner_rust_api_type(&self) -> String {
+        match self {
+            IrTypeDelegate::GeneralArray { general, .. } => general.rust_api_type(),
+            IrTypeDelegate::PrimitiveArray { primitive, .. } => primitive.rust_api_type(),
+            _ => panic!("Unexpected type"),
+        }
+    }
+
+    pub fn inner_is_js_value(&self) -> bool {
+        match self {
+            IrTypeDelegate::GeneralArray { general, .. } => general.is_js_value(),
+            IrTypeDelegate::PrimitiveArray { .. } => false,
+            _ => panic!("Unexpected type"),
+        }
+    }
+
+    pub fn dart_init_method(&self) -> String {
+        match self {
+            IrTypeDelegate::GeneralArray { length, .. } => format!(
+                "{0}.init({1} fill): super(List<{1}>.filled({length},fill));",
+                self.dart_api_type(),
+                self.inner_dart_api_type()
+            ),
+            IrTypeDelegate::PrimitiveArray { length, .. } => format!(
+                "{0}.init(): super({1}({length}));",
+                self.dart_api_type(),
+                self.get_delegate().dart_api_type()
+            ),
+            _ => panic!("Unexpected type"),
+        }
+    }
```

Comment:
Hmm not very sure - should we put several methods to IrTypeDelegate, but indeed it is only reasonable for GeneralArray and PrimitiveArray, while always throw for others

What about this:

```rust
struct IrTypeDelegate {
  Array(IrTypeDelegateArray),
  ... all old things...
}

struct IrTypeDelegateArray {
    GeneralArray {
        length: usize,
        general: Box<IrType>,
    },
    PrimitiveArray {
        length: usize,
        primitive: IrTypePrimitive,
    },
}

impl IrTypeDelegateArray { // NOTE
  pub fn inner_dart_api_type() ... the several methods ...
}
```